src/tree-manager/flatten-tree.js

```diff
@@ -165,13 +180,38 @@ function walkNodes({ nodes, list = new Map(), parent, depth = 0, simple, showPar
         // re-check if all children are checked. if so, check thyself
         if (!isEmpty(node.children) && node.children.every(c => c.checked)) {
           node.checked = true
+          returnValues.checkedNodes.push(node._id)
         }
       }
 
       node.children = undefined
     }
   })
-  return { list, defaultValues }
+
+  return returnValues
+}
+
+function ensureSingleSelect(defaultNodes, checkedNodes) {
```

Comment:
I think we can eliminate this (and the resulting traversal) if we keep track of selection while walking the tree. Something like (pseudo-code) 

```
if(radio && !defaultAlreadyChecked) 
  proceedWithCheckingDefaultValue
```

Reply:
That would not work if a isDefaultValue is first and then a checked node comes along.